api/utxoCache.ts

```diff
@@ -0,0 +1,109 @@
+import { getAddressUtxoOrdinalBundles, getUtxoOrdinalBundle } from './ordinals';
+
+export type UtxoStates = 'inscribed' | 'notInscribed' | 'unknown';
+
+export type UtxoCacheStruct = {
+  [utxoId: string]: UtxoStates;
+};
+
+export type StorageAdapter = {
+  get(key: string): Promise<string | null> | string | null;
+  set(key: string, value: string): Promise<void> | void;
+  remove(key: string): Promise<void> | void;
+};
+
+type UtxoCacheConfig = {
+  cacheStorageController: StorageAdapter;
+};
+
+export class UtxoCache {
+  private readonly _cacheStorageController: StorageAdapter;
+
+  CACHE_STORAGE_KEY = 'utxoCache';
+
+  CACHE_VERSION_STORAGE_KEY = 'utxoCacheVersion';
+
+  VERSION = 1;
+
+  constructor(config: UtxoCacheConfig) {
+    this._cacheStorageController = config.cacheStorageController;
+  }
+
+  _getCache = async (): Promise<UtxoCacheStruct> => {
+    const cache = await this._cacheStorageController.get(this.CACHE_STORAGE_KEY);
+    if (cache) {
+      return JSON.parse(cache);
+    } else {
+      return {};
+    }
+  };
+
+  _getAddressUtxos = async (address: string) => getAddressUtxoOrdinalBundles(address, 0, 200);
```

Comment:
Just a note that the API accepts a max limit of 60 and will truncate to 60 if a higher value is sent.